api/src/main/java/org/apache/iceberg/io/InputFile.java

```diff
@@ -59,4 +60,18 @@
    * @return true if the file exists, false otherwise
    */
   boolean exists();
+
+  // TODO remove this comment after review
```

Comment:
Brainstorming an idea with trait-like solution:
1. Create a new interface for encryption file or reuse the interface `EncryptedInputFile`, which will have method `getNativeDecryptionParameters` and `setNativeDecryptionParameters`. 
2. The subclass(HadoopInputFile, S3InputFile, etc) can implement both `InputFile` and the `EncryptedInputFile`.
3. In case of the caller side, we can invoke it like this  `((EncryptedInputFile)inputFile).getNativeDecryptionParameters()`

Reply:
I like this idea. Regarding the details - using `EncryptedInputFile` directly won't work (because this is a producer/decryptor of input files; an object shouldn't be both an input file and its decryptor) - but we can add a new interface, something like `NativeEncryption`, and make the classes implement it. Moreover, we can apply this to output files, not only to input files. I'll play with this.